src/core/storage/fileio/s3_api.cpp

```diff
@@ -531,10 +550,14 @@ list_objects_response list_objects_impl(s3url parsed_url, std::string proxy,
           } else {
             // continue retry
             std::this_thread::sleep_for(std::chrono::milliseconds(backoff));
-            backoff *= 2;
+            // [2, 4), float random
+            double rdm = (turi::random::pure_random_seed() % 1000 / 500.) + 2;
```

Comment:
I have no idea what `rdm` means. A more descriptive name would be good.

Is this the backoff time in seconds?

Reply:
It means random. The comment above that line doesn't make sense to you?

Reply:
I never would have guessed that `rdm` meant random. So much more time is spent reading code than writing code. So let make variable names descriptive.

You didn't answer my question: is this in seconds?

Reply:
The scale for backoff is ms. This is just a scalar factor.

Reply:
Why do we need a random backoff?  That's only needed when two identical services are competing for the exact same resource, which isn't the rate-limiting service we're talking about here.

Reply:
not exactly the same time. internal aws service blocks subsequent requests within a window time after it decides to throttle the rate. I observed 2 running jobs that start very closely will trigger the rate limiter. Well, let me use regular backoff.